client/my-sites/marketplace/components/billing-interval-switcher/index.tsx

```diff
@@ -56,7 +56,9 @@ const BillingIntervalSwitcher: FunctionComponent< Props > = ( props: Props ) =>
 	const annualLabel = translate( 'Annually' );
 	const saveLabel = translate( 'Save', { context: 'save money' } );
 
-	const searchParams = new URLSearchParams( document.location.search );
+	const searchParams = new URLSearchParams(
+		typeof document !== 'undefined' ? document.location.search : ''
+	);
```

Comment:
Nitpick: What do you think about?

```suggestion
	const searchParams = new URLSearchParams(
		document?.location?.search ?? ''
	);
```

Reply:
This will probably fail (as it did on [this situation](https://github.com/Automattic/wp-calypso/pull/87647/files#r1495731330)) on the server-side as would an arbitrary `undefinedDocument` variable check would be on the client:
<img width="568" alt="Screenshot 2024-04-25 at 08 44 07" src="https://github.com/Automattic/wp-calypso/assets/2749938/d2c94d00-92bf-437b-a0b3-6cb8c1722505">

My hunch is that `document` is assumed `window.document` and the server-side doesn't have `window`. It's likely that `window?.document?.location?.search` would work.